tests/testServer/testServer.cpp

```diff
@@ -1,52 +1,150 @@
+        std::cout << "     Shows this help" << std::endl
+                  << std::endl;
+        std::cout << "Keys/ certs are generated by cert-key-gen.sh and taken from relative path ./cert-key-pair/" << std::endl;
+        std::cout << "BASEPORT:" << std::endl;
+        std::cout << "  The ports used to generate ports for the secure and inscure servers." << std::endl;
+        std::cout << "  Default: 50000" << std::endl;
+        std::cout << "  Example for port generation" << std::endl;
+        std::cout << "  BASEPORT : 50000" << std::endl;
+        std::cout << "  Port for insecure server: 50000 + 51 = 50051" << std::endl;
+        std::cout << "  Port for secure server without client cert: 50000 + 52 = 50052" << std::endl;
+        std::cout << "  Port for secure server with client cert-key pair: 50000 + 443 = 50443" << std::endl;
         return 0;
     }
-    std::string addr = "0.0.0.0:50051";
-    if(argc >=2)
+
+    // Ports withput setting option over cmd
+    //std::string serverAddr = "0.0.0.0:50051";
+    std::string secureServerAddr = "localhost:50443";
+    std::string dfltServerAddr = "localhost:50052";
+    std::string insecureServerAddr = "localhost:50051";
+
+    // Calculation of Ports based on base port specified over cmd
+    if (argc >= 2)
+    {
+        std::string baseport = (argv[1]);
+        int insecurePort = std::stoi(baseport) + 51;
+        int dfltPort = std::stoi(baseport) + 52;
+        int securePort = std::stoi(baseport) + 443;
+
+        insecureServerAddr = "localhost:" + std::to_string(insecurePort);
+        dfltServerAddr = "localhost:" + std::to_string(dfltPort);
+        secureServerAddr = "localhost:" + std::to_string(securePort);
+
+        std::cout << insecurePort << std::endl;
+        std::cout << dfltPort << std::endl;
+        std::cout << securePort << std::endl;
+    }
+
+    std::cout << "Starting secure server with required valid client cert-key pair listening on " << secureServerAddr << std::endl;
+    std::cout << "Starting secure server without required client cert listening on " << dfltServerAddr << std::endl;
+    std::cout << "Starting insecure server listening on " << insecureServerAddr << std::endl;
+
+    // Create a default SSL Credentials object.
+    const char serverKeyPath[] = "cert-key-pair/server_key.pem";
+    const char serverCertPath[] = "cert-key-pair/server_crt.pem";
+    const char clientCertPath[] = "cert-key-pair/client_crt.pem";
+
+    std::shared_ptr<grpc::ServerCredentials> sslCreds;
+    std::shared_ptr<grpc::ServerCredentials> defaultCreds;
+
+    // Get Credentials from Files and define them as  Server Key Cert Pair (needed to fill pem_key_cert_pairs vector of SslCredentialOptions)
+    // Here we can change to chain of trust instead of selfsigned
+    std::string serverKey = gwhisper::util::readFromFile(serverKeyPath);
+    std::cout << "Server: File content of " << serverKeyPath << ": " << serverKey << std::endl;
```

Comment:
do not print whole keys/certs. only print "reading ServerKey from file '/bla/blub/server.key'"